src/ActionModal/components/ActionModal.vue

```diff
@@ -498,20 +498,14 @@ export default {
     selectedBalance() {
       const defaultBalance = {
         amount: 0,
-        gasPrice:
-          this.balances.length > 1
-            ? this.balances.find(({ denom }) => denom === this.getDenom)
-                .gasPrice
-            : 0
+        gasPrice: 4e-7 // the defaultBalance gas price should be the highest we know of to be sure that no transaction gets out of gas
       }
       if (this.balances.length === 0 || !this.network) {
         return defaultBalance
       }
       // default to the staking denom for fees
-      const denom = this.selectedDenom || this.network.stakingDenom
-      let balance = this.balances.find(
-        ({ denom: balanceDenom }) => balanceDenom === denom
-      )
+      const stakingDenom = this.selectedDenom || this.network.stakingDenom
```

Comment:
why the renaming? it's even wrong. this isn't always the stakingDenom

Reply:
oops, true...

But it needs to be renamed because of what I mentioned in my commend above. To do the `denom ===` comparison from the balances array.

Let's call it `preferredDenom` then? Or do you have any suggestion?

Reply:
feeDenom?

Reply:
Done